### ISSUE TITLE
[Feature Request] Stickers, Gifs, Emotes Shared Storage/Sync for Contacts, Chats and Groups

### ISSUE THREAD

**Issue by FlorianSteenbuck** [telegramdesktop/tdesktop#5521]
### Steps to reproduce
1. Create a telegram account
2. Setup 2 Devices
3. Post a gif on Device A
4. Watch in the gif overview for gifs and there is the gif you post on Device A

### Expected behaviour
Everything works fine and this is a cool feature.

### Actual behaviour
But why not implement this per chat or per user in your contacts, with a simple request that you can send to the user to have a synced store of gifs, stickers and emotes.

May also be useful as a group setting (is chat so this can also solve a bot), this would make the plattform attractive for streamers and discord users, they got a similar system with emotes. This create a great opportunity for personalize community's on telegram and on the server side as I known this shouldn't be a so big case just:

**sync_table**

| primary_key | type | foreign_key |
| -- | -- | -- |
| 0 | device | 1856 |
| 2 | chat | 4 |
| 1 | contact | 157 |

(just add a type option to your current system)

### Configuration
**Operating system:**
Linux

**Version of Telegram Desktop:**
1.5.2

**Used theme**:
Dark

**Comment by stek29** [comment]
> this would make the plattform attractive for streamers and discord users

For that you can set a group sticker pack which is  shown to users when they are chatting in that group.

As for emoji, they're not synced at all and I don't think they will ever be synced.

Also, I don't think that sharing frequently used stickers/gifs is a good idea from privacy  perspective.
And finally, this doesn't really fit telegram feature philosophy.

tldr: "streamers" can set group sticker pack for their communities. There's nothing like that for gifs and emoji, and I don't think it'd be a good idea.

Also, it's an API issue -- it has to be implemented on whole platform, not just in one app.
